src/plugins/discover/public/components/data_types/logs/log_level.tsx

```diff
@@ -7,53 +7,43 @@
  */
 
 import React from 'react';
-import { useEuiTheme } from '@elastic/eui';
-import { LogFlyoutDoc } from '@kbn/discover-utils/src';
+import { LogFlyoutDoc, LogLevelBadge } from '@kbn/discover-utils/src';
+import { EuiBadge, EuiBadgeProps } from '@elastic/eui';
 import * as constants from '../../../../common/data_types/logs/constants';
-import { ChipWithPopover } from './popover_chip';
-
-const LEVEL_DICT = {
-  error: 'danger',
-  warn: 'warning',
-  info: 'primary',
-  debug: 'accent',
-} as const;
+import { ChipPopover } from './popover_chip';
 
 interface LogLevelProps {
   level: LogFlyoutDoc['log.level'];
-  dataTestSubj?: string;
-  renderInFlyout?: boolean;
 }
 
-export function LogLevel({ level, dataTestSubj, renderInFlyout = false }: LogLevelProps) {
-  const { euiTheme } = useEuiTheme();
+export function LogLevel({ level }: LogLevelProps) {
   if (!level) return null;
-  const levelColor = LEVEL_DICT[level as keyof typeof LEVEL_DICT]
-    ? euiTheme.colors[LEVEL_DICT[level as keyof typeof LEVEL_DICT]]
-    : null;
-
-  const truncatedLogLevel = level.length > 10 ? level.substring(0, 10) + '...' : level;
-
-  if (renderInFlyout) {
-    return (
-      <ChipWithPopover
-        property={constants.LOG_LEVEL_FIELD}
-        text={truncatedLogLevel}
-        borderColor={levelColor}
-        style={{ width: 'none' }}
-        dataTestSubj={dataTestSubj}
-        shouldRenderPopover={!renderInFlyout}
-      />
-    );
-  }
 
   return (
-    <ChipWithPopover
+    <ChipPopover
       property={constants.LOG_LEVEL_FIELD}
       text={level}
-      rightSideIcon="arrowDown"
-      borderColor={levelColor}
-      style={{ width: '80px', marginTop: '-3px' }}
+      renderChip={({ handleChipClick, handleChipClickAriaLabel, chipCss }) => {
+        const sharedProps: EuiBadgeProps = {
+          iconType: 'arrowDown',
+          iconSide: 'right',
+          onClick: handleChipClick,
+          onClickAriaLabel: handleChipClickAriaLabel,
+          css: [chipCss, { width: '80px', paddingInline: '4px' }],
+        };
+
+        return (
+          <LogLevelBadge
+            {...sharedProps}
+            logLevel={level}
+            fallback={
+              <EuiBadge {...sharedProps} color="hollow">
+                {level}
+              </EuiBadge>
+            }
+          />
```

Comment:
> [!TIP]
> Having the `fallback` property required makes the component usage much more verbose than it needs to be, as we mostly want to display the `hollow` colored version of the badge when there is no matching color.
> I would make the `fallback` property optional and use the following logic to simplify the usage of the component:
> 1. If the color is correctly matched by the logLevel, render the Badge with the matched color.
> 2. If the color is NOT correctly matched by the logLevel and `fallback` is provided, render the fallback.
> 3. If the color is NOT correctly matched by the logLevel, render the Badge with the default `hollow` color.

Ideally, the fallback should be used only if we want to render something different from the badge when the color is not matched.

Wdyt? Is there any scenario I didn't see where is better to have `fallback` required?

Reply:
Good point! It makes sense to only override it when necessary. Updated here: https://github.com/elastic/kibana/pull/188615/commits/136a4c58e3ed063105cd318f81769a1db91f5d9f.